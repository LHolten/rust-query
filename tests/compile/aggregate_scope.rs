use rust_query::{args::Aggregate, Column};

fn columns<'outer, 'inner, S: 'static>(
    outer: Column<'outer, S, i64>,
    inner: Column<'inner, S, i64>,
    aggr: &mut Aggregate<'outer, 'inner, S>,
) {
    aggr.filter_on(&inner, &outer);
    aggr.filter_on(outer, 10);
    aggr.filter_on(10, inner);
}

fn sum<'outer, 'inner, S: 'static>(
    outer: Column<'outer, S, i64>,
    aggr: &Aggregate<'outer, 'inner, S>,
) {
    aggr.sum(outer);
}

fn main() {}
