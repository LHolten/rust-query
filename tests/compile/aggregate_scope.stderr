error: lifetime may not live long enough
 --> tests/compile/aggregate_scope.rs:9:5
  |
3 | fn columns<'outer, 'inner, S: 'static>(
  |            ------  ------ lifetime `'inner` defined here
  |            |
  |            lifetime `'outer` defined here
...
9 |     aggr.filter_on(outer, 10);
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^ argument requires that `'outer` must outlive `'inner`
  |
  = help: consider adding the following bound: `'outer: 'inner`

error: lifetime may not live long enough
  --> tests/compile/aggregate_scope.rs:10:5
   |
3  | fn columns<'outer, 'inner, S: 'static>(
   |            ------  ------ lifetime `'inner` defined here
   |            |
   |            lifetime `'outer` defined here
...
10 |     aggr.filter_on(10, inner);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^ argument requires that `'inner` must outlive `'outer`
   |
   = help: consider adding the following bound: `'inner: 'outer`

help: `'outer` and `'inner` must be the same: replace one with the other

error: lifetime may not live long enough
  --> tests/compile/aggregate_scope.rs:17:5
   |
13 | fn sum<'outer, 'inner, S: 'static>(
   |        ------  ------ lifetime `'inner` defined here
   |        |
   |        lifetime `'outer` defined here
...
17 |     aggr.sum(outer);
   |     ^^^^^^^^^^^^^^^ argument requires that `'outer` must outlive `'inner`
   |
   = help: consider adding the following bound: `'outer: 'inner`
